core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -646,6 +646,13 @@ private boolean isExistingRelation(Session session, QualifiedObjectName name)
             return true;
         }
 
+        if (!getRedirectedTableName(session, name).equals(name)) {
+            // If the table is redirected, that means the source table name is recognized. So, we do not check for existence
```

Comment:
I'm not sure that is always a valid assumption. Do we check for existence of proper table handle in other places that use redirection.
IMO we should do the actual check, because it's cheap and target handle might not be valid

Reply:
I think the point here is return correctly in 
https://github.com/trinodb/trino/blob/1bf4b65d959ccdd2c1c7c08087a8730e4596db21/core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java#L611-L620

we have options

- table that redirects correctly
- table that redirects to non-existent

including both cases in `SHOW TABLES` looks like a correct thing to do, since we cannot throw.

Reply:
Ok, so how do we handle redirections during listing (when no object name is provided)? We should be consistent with that

Reply:
> how do we handle redirections during listing (when no object name is provided)

unfiltered `listTables` doesn't care about redirections, just returns connector's results.

@findepi for the redirected+filtered case, do you mean we should try getting redirected table handle, and return empty if there's an exception doing that?  That'd be closer to current filtered+unredirected case, but different from unfiltered+redirected case.

Reply:
> try getting redirected table handle, and return empty if there's an exception doing that? 


that could be reasonable when we can emit a warning at the same time

for now, i would rather fail instead.
the situation where we _do_ redirect but then get "404", looks like misconfiguration.
we're probably yet to learn what are the conditions where such situation would be legitimate (outside of our control)

Reply:
@findepi does the fixup commit look better?